### PR TITLE
Install local sdk package in doc workflow

### PR DESCRIPTION
**Pull request by lucaato**

### CODE REVIEW

File: .github/workflows/doc-workflow.yaml [hunk 1]
```diff
@@ -46,7 +46,8 @@ jobs:
         working-directory: ./astarte-device-sdk-python/docs
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine requests paho-mqtt cryptography bson PyJWT "Sphinx<7" m2r2 sphinx_rtd_theme
+          pip install "Sphinx<7" m2r2 sphinx_rtd_theme
```
Comment (sorru94) [inline .github/workflows/doc-workflow.yaml:49]: I would place those three packages in a `requirements.txt` in the docs folder. Just to make it cleaner and also being easier to run sphinx locally

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Copyright 2024 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+"Sphinx<7"
```
Comment (sorru94) [inline docs/requirements.txt:4]:
```suggestion
Sphinx<=8
```

Comment (sorru94) [reply docs/requirements.txt:4]: I tested it on my machine, it appears there is no real reason to keep it to <7
So feel free to remove such constraint

Comment (lucaato) [reply docs/requirements.txt]: updated

**Review by sorru94 — APPROVED** [review]

**Pull request merged**